src/physics/initial_conditions/initial_condition.cpp

```diff
@@ -126,6 +126,30 @@ ::value(const dealii::Point<dim,real> &point, const unsigned int /*istate*/) con
 
 }
 
+// ========================================================
+// 1D SINE -- Initial Condition for advection_explicit_time_study
+// ========================================================
+template <int dim, int nstate, typename real>
+InitialConditionFunction_1DSine<dim,nstate,real>
+::InitialConditionFunction_1DSine ()
+        : InitialConditionFunction<dim,nstate,real>()
+{
+    // Nothing to do here yet
+}
+
+template <int dim, int nstate, typename real>
+inline real InitialConditionFunction_1DSine<dim,nstate,real>
+::value(const dealii::Point<dim,real> &point, const unsigned int /*istate*/) const
+{
+    real value = 0;
+    real pi = dealii::numbers::PI;
+    if(point[0] >= 0.0 && point[0] <= 2.0){
+        value = sin(2*pi*point[0]/2.0);
```

Comment:
if this is essentially the respective exact solution at `t=0`, consider calling it here and passing `t=0`

Reply:
Yes, it is. I think that falls under the discussion of should we unify exact solution and initial condition.